dmci/api/worker.py

```diff
@@ -87,30 +87,60 @@ def validate(self, data):
         # Check xml file against XML schema definition
         valid = self._xsd_obj.validate(etree.fromstring(data))
         msg = repr(self._xsd_obj.error_log)
-        data_mod = data
+
         if valid:
             # Check information content
             valid, msg = self._check_information_content(data)
-            # Append env string to namespace in data
+            if not valid:
+                return valid, msg, data
+
             if self._conf.env_string:
+
+                # Append env string to namespace in metadata_identifier
                 logger.debug("Identifier namespace: %s" % self._namespace)
                 logger.debug("Environment customization %s" % self._conf.env_string)
-                ns_re_pattern = re.compile(r"\w.\w."+self._conf.env_string)
+                ns_re_pattern = re.compile("\\w.\\w."+self._conf.env_string)
```

Comment:
Why not use the r-formatting?

Reply:
I do not remember why I had changed this, r-formatting should be fine...